src/services/user.service.ts

```diff
@@ -10,8 +17,93 @@ export class UserService {
     private readonly userRepository: Repository<User>,
   ) {}
 
+  async get(param: string): Promise<UserResponseDto[]> {
+    const users = await this.userRepository.find({
+      relations: { role: true, appointments: true },
+    });
+
+    const userResponse = users.map((user) => {
```

Comment:
is it possible to do something like?:

```suggestion
const userResponse = users.map(user => new UserResponseDto(
  user.id,
  user.name,
  user.phone,
  user.email,
  user.role,
  user.appointments
));
```

Reply:
or: 

```suggestion
const userResponse = users.map(user => ({
  ...new UserResponseDto(),
  id: user.id,
  name: user.name,
  phone: user.phone,
  email: user.email,
  role: user.role,
  appointments: user.appointments,
}));
```

Reply:
The first is not possible because the `UserResponseDto` class does not define a constructor that accepts arguments and would have to explicitly define a constructor in my dto.

The second one is way better, but how I did it is how it is usually done in nestjs, so the second one is not wrong, but is not the best solution as well

Reply:
got it, thanks for clarifying!